### PR TITLE
ARROW-17831: [Python][Docs] PyArrow Architecture page outdated after moving pyarrow C++ code

### PR DESCRIPTION
**Pull request by AlenkaF**

### CODE REVIEW

File: docs/source/python/getting_involved.rst [hunk 1]
```diff
@@ -70,6 +70,6 @@ used as foundations to build easier to use entities.
   so that in the other ``.pyx`` files they can be used to implement Python classes,
   functions and helpers.
 * PyArrow is also based on dedicated pieces of code that are written into Arrow itself.
```
Comment (jorisvandenbossche) [inline docs/source/python/getting_involved.rst:72]: I think this sentence can also be updated. While we of course still depend on "Arrow" (meaning here the C++ library, although that should probably be made explicit), it is not how this was meant now this code lives in pyarrow itself.

Comment (AlenkaF) [reply docs/source/python/getting_involved.rst:72]: Yes, agree. Will correct.

File: docs/source/python/getting_involved.rst [hunk 2]
```diff
@@ -47,7 +47,7 @@ less complex to use. So while in some cases it might be easy to map what's
 in C++ to what's in Python, in many cases the C++ classes and methods are
 used as foundations to build easier to use entities.
 
-.. image:: py_arch_overview.svg
+.. image:: /python/savefig/py_arch_overview.svg
```
[the remark this answers is not in the document]
Comment (AlenkaF) [reply docs/source/python/getting_involved.rst:50]: Oh, I was sure this folder existed before! 🤦‍♀️  I guess it is created when building the docs or something similar.
Yes, I can use `images/` or just leave it in `python/` folder as before.

File: docs/source/python/getting_involved.rst (hunk 1 above)
Comment (AlenkaF) [reply docs/source/python/getting_involved.rst]: > Apart from Arrow C++ library, which dependence is mentioned in the previous line, PyArrow is also based on PyArrow C++, dedicated pieces of code that live in ``python/pyarrow/src/arrow/python`` directory and provide the low level ...

What do you think of the change?

File: docs/source/python/getting_involved.rst (hunk 2 above)
Comment (jorisvandenbossche) [inline docs/source/python/getting_involved.rst:50]:
```suggestion
.. image:: /python/images/py_arch_overview.svg
```

(you renamed the file, but not changed it here, I think)

**Comment by jorisvandenbossche** [comment]
This can be undrafted now https://github.com/apache/arrow/pull/14275 is merged?

File: docs/source/python/getting_involved.rst (hunk 2 above)
Comment (AlenkaF) [reply docs/source/python/getting_involved.rst:50]: 🤦‍♀️

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request merged**